opencti-platform/opencti-graphql/src/database/stix-2-0-converter.ts

```diff
@@ -0,0 +1,83 @@
+
+// Meta
+const buildKillChainPhases = (instance: StoreEntity | StoreRelation): Array<SMO.StixInternalKillChainPhase> => {
+  return (instance[INPUT_KILLCHAIN] ?? []).map((k) => {
+    const data: SMO.StixInternalKillChainPhase = {
+      kill_chain_name: k.kill_chain_name,
+      phase_name: k.phase_name,
+      x_opencti_order: k.x_opencti_order,
+    };
+    return cleanObject(data);
+  });
+};
+
+const buildExternalReferences = (instance: StoreObject): Array<SMO.StixInternalExternalReference> => {
+  return (instance[INPUT_EXTERNAL_REFS] ?? []).map((e) => {
+    const data: SMO.StixInternalExternalReference = {
+      source_name: e.source_name,
+      description: e.description,
+      url: e.url,
+      hash: e.hashes, // key is "hash" in client python but should be "hashes"
+      external_id: e.external_id,
+    };
+    return cleanObject(data);
+  });
+};
+
+// Builders
+const buildStixObject = (instance: StoreObject): S.StixObject => {
+  return {
+    id: instance.standard_id,
+    x_opencti_id: instance.id,
+    spec_version: '2.0',
+    x_opencti_type: ENTITY_TYPE_MALWARE,
+    type: instance.entity_type.toLowerCase(),
+    x_opencti_granted_refs: (instance[INPUT_GRANTED_REFS] ?? []).map((m) => m.standard_id),
+    x_opencti_workflow_id: instance.x_opencti_workflow_id,
+  };
+};
+
+// General
+const buildStixDomain = (instance: StoreEntity | StoreRelation): S.StixDomainObject => {
+  return {
+    ...buildStixObject(instance),
+    created: convertToStixDate(instance.created),
+    modified: convertToStixDate(instance.modified),
+    revoked: instance.revoked,
+    confidence: instance.confidence,
+    // lang: instance.lang,
+    labels: (instance[INPUT_LABELS] ?? []).map((m) => m.value), // memo : x_opencti_labels for SCO
+    object_marking_refs: (instance[INPUT_MARKINGS] ?? []).map((m) => m.standard_id),
+    created_by_ref: instance[INPUT_CREATED_BY]?.standard_id, // memo: x_opencti_created_by_ref for SCO
+    external_references: buildExternalReferences(instance),
+  };
+};
+
+export const convertMalwareToStix2 = (instance: StoreEntity, type: string): SDO.StixMalware => {
+  assertType(ENTITY_TYPE_MALWARE, type);
+  return {
+    ...buildStixDomain(instance),
+    name: instance.name,
+    description: instance.description,
+    malware_types: instance.malware_types,
+    is_family: instance.is_family,
+    aliases: instance.aliases,
+    kill_chain_phases: buildKillChainPhases(instance),
+    first_seen: convertToStixDate(instance.first_seen),
+    last_seen: convertToStixDate(instance.last_seen),
+    architecture_execution_envs: instance.architecture_execution_envs,
+    implementation_languages: instance.implementation_languages,
+    capabilities: instance.capabilities,
+    operating_system_refs: (instance[INPUT_OPERATING_SYSTEM] ?? []).map((m) => m.standard_id),
+    sample_refs: (instance[INPUT_SAMPLE] ?? []).map((m) => m.standard_id),
+    samples: (instance[INPUT_SAMPLE] ?? []).map((m) => { return { id: m.id, }; }),
```

Comment:
Must be x_opencti_sample_refs_ids i think

Do you take the time to compare your format with the result of the stix file exporter?
Not sure but maybe some elements are missing

Reply:
Yes I compare with actual export-stix-file (see test` malware-stix-2-0-converter-test.ts`)

this is actual STIX export, we do  have  `sample_refs`  : 
![image](https://github.com/user-attachments/assets/1b51aee4-fc91-4577-a2a0-0fed16bd1e39)

Reply:
Thats an error in the python converter. For me sample in 2.0 must no exists like this.
It should be like workflow and follow the pattern rule and so be x_opencti_sample_refs_ids

Reply:
just to be sure @richard-julien  :
`sample_refs` actual attribute should become `x_opencti_sample_refs_ids` ? 
we currently have `samples` (id) and `sample_refs` (standard id). should we keep both except that `sample_refs` would become  `x_opencti_sample_refs_ids`

Reply:
No, sample_refs must remains the same.

if we can samples must be removed. There is no reason to have this attribute.

Reply:
Im not agree with this. Thats needs to be change.

Reply:
Please remove samples and also x_opencti_sample_refs_ids (i finally think its not needed). Thanks